### PR TITLE
compressed plot support

### PR DESCRIPTION
**Pull request by arvidn**
bumps the `chiapos` dependency and adds support for compressed plots to:
1. the harvester
2. chia plots check
3. plot manager

### CODE REVIEW

**Review by AmineKhaldi — APPROVED** [review]

File: chia/harvester/harvester.py [hunk 1]
```diff
@@ -80,6 +81,30 @@ def __init__(self, root_path: Path, config: Dict[str, Any], constants: Consensus
...
+                thread_count,
+                disable_cpu_affinity,
+                max_compression_level_allowed,
+                use_gpu_harvesting,
+                gpu_index,
+                enforce_gpu_index,
+            )
+        except Exception as e:
+            self.log.error(f"{type(e)} {e} while configuring decompresser.")
+            raise
```
Comment (arvidn) [inline chia/harvester/harvester.py:106]: these lines have no test coverage

File: chia/harvester/harvester_api.py [hunk 2]
```diff
@@ -137,6 +137,22 @@ def blocking_lookup(filename: Path, plot_info: PlotInfo) -> List[Tuple[bytes32,
...
+                                        f"File: {filename} Plot ID: {plot_id.hex()}, challenge: {sp_challenge_hash}, "
+                                        f"plot_info: {plot_info}"
+                                    )
+                                else:
+                                    self.harvester.log.error(f"Exception fetching full proof for {filename}. {e}")
+                                    self.harvester.log.error(
+                                        f"File: {filename} Plot ID: {plot_id.hex()}, challenge: {sp_challenge_hash}, "
+                                        f"plot_info: {plot_info}"
+                                    )
+                                continue
```
Comment (arvidn) [inline chia/harvester/harvester_api.py:155]: this entire block has no coverage

File: chia/plotting/check_plots.py
Comment (arvidn) [inline chia/plotting/check_plots.py:1]: we don't run `chia plots check` in CI, so there's no coverage of this either

File: chia/plotting/manager.py [hunk 3]
```diff
@@ -66,13 +67,47 @@ def __init__(
...
+        gpu_index: int,
+        enforce_gpu_index: bool,
+    ) -> None:
+        if max_compression_level_allowed > 7:
+            log.error(
+                "Currently only compression levels up to 7 are allowed, "
+                f"while {max_compression_level_allowed} was specified."
+                "Setting max_compression_level_allowed to 7."
+            )
+            max_compression_level_allowed = 7
```
Comment (arvidn) [inline chia/plotting/manager.py:94]: we don't test the case where we set compression level > 7

File: chia/plotting/manager.py [hunk 4]
```diff
@@ -66,13 +67,47 @@ def __init__(
...
+            max_compression_level_allowed,
+            use_gpu_harvesting,
+            gpu_index,
+            enforce_gpu_index,
+        )
+        if not is_using_gpu and use_gpu_harvesting:
+            log.error(
+                "GPU harvesting failed initialization. "
+                f"Falling back to CPU harvesting: {context_count} decompressers count, {thread_count} threads."
+            )
```
Comment (arvidn) [inline chia/plotting/manager.py:108]: we have no test for this failure mode

File: chia/plotting/manager.py [hunk 5]
```diff
@@ -280,10 +315,20 @@ def process_file(file_path: Path) -> Optional[PlotInfo]:
...
+                    level = prover.get_compression_level()
+                    if level == 0:
+                        if prover.get_size() >= 30 and stat_info.st_size < 0.98 * expected_size:
+                            log.warning(
+                                f"Not farming plot {file_path}. "
+                                f"Size is {stat_info.st_size / (1024 ** 3)} GiB, "
+                                f"but expected at least: {expected_size / (1024 ** 3)} GiB. "
+                                "We assume the file is being copied."
+                            )
+                            return None
```
Comment (arvidn) [inline chia/plotting/manager.py:327]: there's no test coverage for truncated plot files

File: chia/plotting/manager.py [hunk 6]
```diff
@@ -280,10 +315,20 @@ def process_file(file_path: Path) -> Optional[PlotInfo]:
                     # TODO: consider checking if the file was just written to (which would mean that the file is still
                     # being copied). A segfault might happen in this edge case.
 
-                    if prover.get_size() >= 30 and stat_info.st_size < 0.98 * expected_size:
+                    level = prover.get_compression_level()
+                    if level == 0:
```
Comment (arvidn) [inline chia/plotting/manager.py:319]: there's no test coverage of the case `level` != 0

**Comment by arvidn** [comment]
I think it could make sense to declare the lack of test coverage OK, since we're doing extensive manual testing of this. Also since we don't have CI runners that can run plot decompression at all, it makes it a bit trickier

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**